flexmeasures/api/v3_0/sensors.py

```diff
@@ -69,7 +70,11 @@ class SensorAPI(FlaskView):
             "account": AccountIdField(
                 data_key="account_id", load_default=AccountIdField.load_current
```

Comment:
I think I saw a flaw in our design here. I would suggest we only select the current users' account when they have neither specified account nor asset.

Now we always set account to the current user's account if they don't specify, and that might lead to surprising error messages if the user is outside the asset account (e.g. they are an admin or consultant).

We are not allowed to specify an asset that is not in the account (e.g. lilne 192), and that is still okay. I just want to leave the possibility open for account to be `None`. We only set it if we have no account or asset to go on.

Reply:
I can also see the potential issue this could cause, especially in cases involving admin or consultant roles.

Got it. So now, the default will load as None, and we’ll have an additional check to set the account to the current user’s account if both account and asset are None.

Reply:
we will then have the below in the top part of the function

```
if account and asset is None:
            account = current_user.account if not current_user.is_anonymous else None
       ```

Reply:
So we will add the below logic

```
        if account and asset is None:
            account = current_user.account if not current_user.is_anonymous else None

        account_ids: list = [account.id] if account else []
```

Reply:
I guess you meant if account is None right?

Reply:
This part will have to be moved into the code as the account can be None now and this will return a 403

`@permission_required_for_context("read", ctx_arg_name="account")`

Reply:
> I guess you meant if account is None right?

Yes, the bottom part s for a scenario where asset I is provided but account was not

Reply:
I've been reviewing this logic for a while, and I'm a bit unclear on the expected behavior. In a case where no account ID is provided but an asset ID is passed, what should the intended outcome be?

Reply:
I guess you want to act on the situation where neither account nor asset is set, and in that case pick current_user.account, right?

Does this work?

```
if account is None and asset is None:  # we default to the current user's account
    if current_user.is_anonymous:
        raise Unauthorized
    check_access(account, "read")
    account = current_user.account
```